Tutorials/json-tutorials/11-01-json_reading.mdx

```diff
@@ -0,0 +1,96 @@
+---
+title: Reading JSON Data in SplashKit
+description: After understanding the basics of JSON in SplashKit, this part of the tutorial focuses on how to read and parse JSON data. Reading JSON data is essential for game development tasks such as loading game settings, level configurations, or player data.
+category: Guides
+author: Jonathan Tynan
+lastupdated: May 11 2024
+---
+
+import { Tabs, TabItem } from "@astrojs/starlight/components";
+
+**{frontmatter.description}**
+
+_Written by {frontmatter.author} on {frontmatter.lastupdated}_
+
+## Reading JSON Objects
+
+
+
+In the previous tutorial we loaded the following [JSON](/api/json/) file and read the game title from it. Lets extend this a little, and dive a further into extracting values from this structure.
+
+```json
+{
+  "gameTitle": "My New Game",
+  "fullScreenMode": false,
+  "numPlayers": 1,
+  "screenSize": {
+    "width": 800,
+    "height": 600
+  },
+  "levels": ["level1", "level2", "level3"]
+}
+```
+
+### Accessing Values
+
+To access values in JSON objects like strings, numbers, or booleans, you can use functions like [`json_read_string()`](/api/json/#json-read-string), [`json_read_number_as_int()`](/api/json/#json-read-number-as-int), or [`json_read_bool()`](/api/json/#json-read-bool). We use these functions like the following code snippet.
+
+<Tabs>
+    <TabItem label="C++">
+    ```cpp
+    string title = json_read_string(game_data, "gameTitle");
+    int numPlayers = json_read_number_as_int(game_data, "numPlayers");
+    bool isFullScreen = json_read_bool(game_data, "fullScreenMode");
+    ```
+    </TabItem>
+</Tabs>
+
+### Working with JSON Arrays
+
+If the data is an array, like the value stored for the `levels` key, we can obtain the data through [`json_read_array()`](/api/json/#json-read-array) and store it into a vector. Then we can loop through that vector and do some action with the stored data. Below is an example of this.
+
+<Tabs>
+    <TabItem label="C++">
+    ```cpp
+    vector<string> levels;
+    json_read_array(game_data, "levels", levels);
+    int num_levels = levels.size();
```

Comment:
maybe just chuck in a comment for .size() being a built in function returning the # of elements, the code is generally explained before what you're doing so comments can become superfluous, can also just ignore this suggestion if it comes off as nitpicky